Cleanup and Fix

Road-hog123 opened the pull request:
- The syntax files were indented with tabs rather than spaces.
- Many files were missing trailing newlines.
- A couple of files had mixed or trailing whitespace.
- Some files used CRLF - a .gitattributes has been added to prevent this happening again.
- Two omsiscript scopes were incorrect.
- omsiscript numeric constants only matched at the beginning of a line, and were permitted to have nothing after the decimal point.
- The $d operator was not detected.
- Several regexes with repetition have been adjusted to reduce that repetition.

Road-hog123 committed:
Cleanup whitespace

Road-hog123 committed:
Add .gitattributes

Fix mixed eol
Exclude .gitignore from git archive

Road-hog123 committed:
Ignore vsix packages

Road-hog123 committed:
Fix trailing comma

Road-hog123 committed:
OMSI Script does not use brackets

Included snippets also remove the need for auto closing and surrounding

Road-hog123 committed:
Fix incorrect scopes

Road-hog123 committed:
Fix and cleanup regexes

Numeric constants only matched at the start of a line and could have
nothing after the decimal point

$d was not matched

Repetition in patterns has been reduced

fuljo reviewed (commented):
As always, thank you very much for your dedication to this project, I really appreciate it. Everything is fine, aside from a couple of thoughts I had on parenthesization.

I agree to remove '{', '[' and '(' from brackets, as they are not really used to enclose an expression.
However I was thinking about keeping '{' and '(' as auto-closing and surrounding pairs, since they are used in macro/trigger definitions and variable access respectively.
Still, I haven't been writing OMSI scripts for a while, and I fear that having auto-closing pairs for  '(' might be annoying when writing code. Please tell me your thoughts on this.

Road-hog123 committed:
Keep auto-closing and auto-surrounding brackets

Auto-closing is disabled within strings, auto-surrounding does not
support this feature.

Road-hog123 commented:
I'm perfectly happy to keep them, I can see that others may find them useful in their workflows. I've disabled auto-closing them within strings, as I can see it being likely they would be on their own there, but unfortunately vscode does not support doing the same for auto-surrounding.

vscode makes it very easy to change the indentation style in a file, so if it would be preferred to use something other than 4 spaces I can easily do that too.
